src/core/ccitt_stream.js

```diff
@@ -0,0 +1,71 @@
+/* Copyright 2012 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Dict, isDict } from './primitives';
+import { CCITTFaxDecoder } from './ccitt';
+import { DecodeStream } from './stream';
+import { shadow } from '../shared/util';
+
+var CCITTFaxStream = (function CCITTFaxStreamClosure() {
+  function CCITTFaxStream(str, maybeLength, params) {
+    this.str = str;
+    this.dict = str.dict;
+    this.maybeLength = maybeLength;
+    this.params = isDict(params) ? params : Dict.empty;
+
+    DecodeStream.call(this, maybeLength);
+  }
+
+  CCITTFaxStream.prototype = Object.create(DecodeStream.prototype);
+
+  Object.defineProperty(CCITTFaxStream.prototype, 'bytes', {
+    get() {
+      // If this.maybeLength is null, we'll get the entire stream.
+      return shadow(this, 'bytes', this.str.getBytes(this.maybeLength));
+    },
+    configurable: true,
+  });
+
+  CCITTFaxStream.prototype.ensureBuffer = function() {
+    if (this.bufferLength) {
+      return;
+    }
+
+    let ccittFaxDecoder = new CCITTFaxDecoder({
+      K: this.params.get('K'),
+      EndOfLine: this.params.get('EndOfLine'),
+      EncodedByteAlign: this.params.get('EncodedByteAlign'),
+      Columns: this.params.get('Columns'),
+      Rows: this.params.get('Rows'),
+      EndOfBlock: this.params.get('EndOfBlock'),
+      BlackIs1: this.params.get('BlackIs1'),
+    });
+    let data = ccittFaxDecoder.parse(this.bytes, this.maybeLength);
```

Comment:
Let's change this to pass a "source" of data, and this interface will have a single next() methods (which will return -1 for EOF).

Reply:
> Let's change this to pass a "source" of data,

*Hopefully I didn't completely misunderstand what you're asking for here :-)*

I've change the interface of `CCITTFaxDecoder` such that you can either initialize `CCITTFaxDecoder` with a stream (used from `CCITTFaxStream`), or pass in a `Uint8Array` instead (which seems like what you'd want to do in PR #8991 when decoding Huffman data).

> and this interface will have a single next() methods

Used the `readNextChar` name (as suggested above).

> (which will return -1 for EOF).

Good point, fixed now!